Fix and refactor CreateBulkConnections_ConfCons2_6.ps1

sk82jack opened the pull request:
## Description
The bulk connections PowerShell script currently doesn't work as it references non-existent classes (presumably from an old class structure). This PR fixes the issue by using the new classes. A lot of the script has been re-written to adhere to PowerShell best practices and to paramterize a lot of the .Net methods. The changes are:
- Remove trailing whitespaces per the [PowerShell community best practices](https://github.com/PoshCode/PowerShellPracticeAndStyle/blob/master/Style-Guide/Code-Layout-and-Formatting.md#trailing-spaces).
- Query the registry for the install path instead of requiring the end user to edit the script
- Update the syntax for instantiating objects from .Net classes from the old `New-Object` syntax.
- Rename functions using a shorter naming convention
- Add parameters and logic to functions to avoid instantiating .Net classes and calling .Net methods in the examples
- Make use of the FileDataProvider class to export the serialized XML instead of building it manually as a string
- Flesh out Example 2 to provide an example of nested folders and the use of the PassThru parameter to edit connection properties

## Motivation and Context
This fix will enable users to download the bulk connections PowerShell script and run it without having to troubleshoot why it's throwing errors.
Fixes #1149

## How Has This Been Tested?
It's been tested on a Windows 10 1809 PowerShell 5.1 machine and it produces the XML without errors for both examples. 
As this is an example/documentation change it doesn't affect other areas of the code.

## Types of changes
<!--- What types of changes does your code introduce? Put an `x` in all the boxes that apply: -->
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.

sparerd commented on Tools/CreateBulkConnections_ConfCons2_6.ps1:
```diff
@@ -1,114 +1,237 @@
-
-if ($PathToMrngFolder -eq "") {
-    Write-Error -Message 'You must set the $PathToMrngFolder variable in this script to the folder which contains mRemoteNG.exe'
+foreach ($Path in 'HKLM:\SOFTWARE\WOW6432Node\mRemoteNG', 'HKLM:\SOFTWARE\mRemoteNG') {
```
This will work for installed versions of mRemoteNG, but portable versions will not have a registry entry. If no version was installed on the machine, the later assembly load will fail. Could a prompt be added if no reg entry is found?

sk82jack replied:
Ah didn't realise you had a portable version! Yeah, I'll add a prompt for if no installed version is found 🙂 Do you want me to squash the commits to tidy them up or you OK to have minor fixes in the commit history?

sparerd replied:
We are ok with minor fixes like that, no need to squash.

sparerd commented:
Thanks for taking the time to update the script @sk82jack. Things look good, I've just made one comment to address in the code

sparerd commented on Tools/CreateBulkConnections_ConfCons2_6.ps1:
```diff
@@ -1,114 +1,237 @@
+#   Replace or modify the examples that are shown toward the end of the script to create your own connection info objects.
 #####################################
 
-$EncryptionKey = (Get-Credential -Message "Enter the encryption key you would like to use. This must match the encryption key used by the rest of the confCons file." -UserName "DontNeedUsername").Password
```
Its only really necessary if a value needs to be encrypted. If any of the password fields are empty, are inherited from a parent node, or filtered out through the SaveFilter, then no encryption happens (we write an empty string).

My guess is that in your case the password field was empty.

sk82jack commented:
Apologies for the delay with this - been a busy couple of weeks!

I added a folder selection for a portable installation as requested. I also added a check for whether a password property has been set and if so, prompted for the decryption password.

sparerd commented:
Looks good, thanks for taking the time to update this script @sk82jack!
